---
layout: page
title: PH525x series - Biomedical Data Science
---

## Links and resources

- [R markdown source files](http://github.com/genomicsclass/labs)
- [ePub version on Leanpub](https://leanpub.com/dataanalysisforthelifesciences/)
- [Links to the HarvardX class pages](pages/classes.html)
- [External resources and books](pages/resources.html)
- [Finding more help for data analysis](pages/more_help.html)


## Chapter 0 - Introduction
- [Introduction](pages/introduction.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/intro/introduction.Rmd)
- [Getting started](pages/getting_started.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/intro/getting_started.Rmd)
- [Getting started exercises](pages/getting_started_exercises.html)
- [dplyr introduction](pages/dplyr_intro.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/intro/dplyr_intro.Rmd)
- [dplyr introduction exercises](pages/dplyr_intro_exercises.html)
- [Mathematical notation](pages/math_notation.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/intro/math_notation.Rmd)

## Chapter 1 - Inference
- [Random variables](pages/random_variables.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/inference/random_variables.Rmd)
- [Random variables exercises](pages/random_variables_exercises.html)
- [Populations and samples](pages/populations_and_samples.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/inference/populations_and_samples.Rmd)
- [Populations and samples exercises](pages/populations_and_samples_exercises.html)
- [CLT and t-distribution](pages/clt_and_t-distribution.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/inference/clt_and_t-distribution.Rmd)
- [CLT and t-distribution exercises](pages/clt_and_t-distribution_exercises.html)
- [CLT in practice](pages/clt_in_practice.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/inference/clt_in_practice.Rmd)
- [CLT in practice exercises](pages/clt_in_practice_exercises.html)
- [t-test in practice](pages/t-tests_in_practice.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/inference/t-tests_in_practice.Rmd)
- [Confidence intervals](pages/confidence_intervals.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/inference/confidence_intervals.Rmd)
- [Power calculations](pages/power_calculations.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/inference/power_calculations.Rmd)
- [Power calculations exercises](pages/power_calculations_exercises.html)
- [Monte carlo](pages/monte_carlo.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/inference/monte_carlo.Rmd)
- [Monte carlo exercises](pages/monte_carlo_exercises.html)
- [Permutation tests](pages/permutation_tests.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/inference/permutation_tests.Rmd)
- [Permutation tests exercises](pages/permutation_tests_exercises.html)
- [Association tests](pages/association_tests.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/inference/association_tests.Rmd)
- [Association tests exercises](pages/association_tests_exercises.html)

## Chapter 2 - Exploratory Data Analysis
- [Exploratory data analysis](pages/exploratory_data_analysis.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/eda/exploratory_data_analysis.Rmd)
- [Plots to avoid](pages/plots_to_avoid.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/eda/plots_to_avoid.Rmd)
- [Exploratory data analysis exercises](pages/exploratory_data_analysis_exercises.html)

## Chapter 3 - Robust Statistics
- [Robust summaries](pages/robust_summaries.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/robust/robust_summaries.Rmd)
- [Rank tests](pages/ranktest.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/robust/ranktest.Rmd)
- [Robust summaries exercises](pages/robust_summaries_exercises.html)

## Chapter 4 - Matrix Algebra
- [Introduction to using regression](pages/intro_using_regression.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/matrixalg/intro_using_regression.Rmd)
- [Introduction to using regression exercises](pages/intro_using_regression_exercises.html)
- [Matrix notation](pages/matrix_notation.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/matrixalg/matrix_notation.Rmd)
- [Matrix notation exercises](pages/matrix_notation_exercises.html)
- [Matrix operations](pages/matrix_operations.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/matrixalg/matrix_operations.Rmd)
- [Matrix operations exercises](pages/matrix_operations_exercises.html)
- [Matrix algebra examples](pages/matrix_algebra_examples.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/matrixalg/matrix_algebra_examples.Rmd)
- [Matrix algebra examples exercises](pages/matrix_algebra_examples_exercises.html)

## Chapter 5 - Linear Models
- [Linear models introduction](pages/linear_models_intro.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/linear/linear_models_intro.Rmd)
- [Linear models introduction exercises](pages/linear_models_intro_exercises.html)
- [Expressing design formula](pages/expressing_design_formula.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/linear/expressing_design_formula.Rmd)
- [Expressing design formula exercises](pages/expressing_design_formula_exercises.html)
- [Linear models in practice](pages/linear_models_in_practice.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/linear/linear_models_in_practice.Rmd)
- [Linear models in practice exercises](pages/linear_models_in_practice_exercises.html)
- [Standard errors](pages/standard_errors.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/linear/standard_errors.Rmd)
- [Standard errors exercises](pages/standard_errors_exercises.html)
- [Interactions and contrasts](pages/interactions_and_contrasts.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/linear/interactions_and_contrasts.Rmd)
- [Interactions and contrasts exercises](pages/interactions_and_contrasts_exercises.html)
- [Collinearity](pages/collinearity.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/linear/collinearity.Rmd)
- [Collinearity exercises](pages/collinearity_exercises.html)
- [QR and regression](pages/qr_and_regression.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/linear/qr_and_regression.Rmd)
- [Linear models going further](pages/linear_models_going_further.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/linear/linear_models_going_further.Rmd)

## Chapter 6 - Inference for High-Dimensional Data
- [Introduction to high-throughput data](pages/intro_to_highthroughput_data.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/advinference/intro_to_highthroughput_data.Rmd)
- [Introduction to high-throughput data exercises](pages/intro_to_highthroughput_data_exercises.html)
- [Inference for high-throughput data](pages/inference_for_highthroughput.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/advinference/inference_for_highthroughput.Rmd)
- [Inference for high-throughput data exercises](pages/inference_for_highthroughput_exercises.html)
- [Multiple testing](pages/multiple_testing.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/advinference/multiple_testing.Rmd)
- [Multiple testing exercises](pages/multiple_testing_exercises.html)
- [EDA for high-throughput data](pages/eda_for_highthroughput.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/advinference/eda_for_highthroughput.Rmd)
- [EDA for high-throughput data exercises](pages/eda_for_highthroughput_exercises.html)

## Chapter 7 - Statistical Modeling
- [Modeling](pages/modeling.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/modeling/modeling.Rmd)
- [Modeling exercises](pages/modeling_exercises.html)
- [Bayes theorem](pages/bayes.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/modeling/bayes.Rmd)
- [Bayes theorem exercises](pages/bayes_exercises.html)
- [Hierarchical models](pages/hierarchical_models.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/modeling/hierarchical_models.Rmd)
- [Hierarchical models exercises](pages/hierarchical_models_exercises.html)

## Chapter 8 - Distance and Dimension Reduction
- [Distance](pages/distance.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/highdim/distance.Rmd)
- [Distance exercises](pages/distance_exercises.html)
- [PCA motivation](pages/pca_motivation.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/highdim/pca_motivation.Rmd)
- [SVD](pages/svd.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/highdim/svd.Rmd)
- [SVD exercises](pages/svd_exercises.html)
- [Projections](pages/projections.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/highdim/projections.Rmd)
- [Rotations](pages/rotations.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/highdim/rotations.Rmd)
- [MDS](pages/mds.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/highdim/mds.Rmd)
- [MDS exercises](pages/mds_exercises.html)
- [PCA](pages/PCA.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/highdim/PCA.Rmd)
- [PCA and SVD in R](pages/pca_svd.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/advinference/storage/pca_svd.Rmd)


## Chapter 9 - Practical Machine Learning
- [Clustering and heatmaps](pages/clustering_and_heatmaps.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/ml/clustering_and_heatmaps.Rmd)
- [Clustering and heatmaps exercises](pages/clustering_and_heatmaps_exercises.html)
- [Conditional expectation](pages/conditional_expectation.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/ml/conditional_expectation.Rmd)
- [Conditional expectation exercises](pages/conditional_expectation_exercises.html)
- [Smoothing](pages/smoothing.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/ml/smoothing.Rmd)
- [Smoothing exercises](pages/smoothing_exercises.html)
- [Machine learning](pages/machine_learning.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/ml/machine_learning.Rmd)
- [Crossvalidation](pages/crossvalidation.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/ml/crossvalidation.Rmd)
- [Crossvalidation exercises](pages/crossvalidation_exercises.html)

## Chapter 10 - Batch Effects
- [Introduction to batch effects](pages/intro_to_batch_effects.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/batch/intro_to_batch_effects.Rmd)
- [Confounding](pages/confounding.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/batch/confounding.Rmd)
- [Confounding exercises](pages/confounding_exercises.html)
- [EDA with PCA](pages/eda_with_pca.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/batch/eda_with_pca.Rmd)
- [EDA with PCA exercises](pages/eda_with_pca_exercises.html)
- [Adjusting with linear models](pages/adjusting_with_linear_models.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/batch/adjusting_with_linear_models.Rmd)
- [Adjusting with linear models exercises](pages/adjusting_with_linear_models_exercises.html)
- [Factor analysis](pages/factor_analysis.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/batch/factor_analysis.Rmd)
- [Factor analysis exercises](pages/factor_analysis_exercises.html)
- [Adjusting with factor analysis](pages/adjusting_with_factor_analysis.html)  [[Rmd]](https://github.com/genomicsclass/labs/blob/master/batch/adjusting_with_factor_analysis.Rmd)
- [Adjusting with factor analysis exercises](pages/adjusting_with_factor_analysis_exercises.html)

---

<a name="5xtop"></a>

## Chapter 11 - Introduction to Bioconductor

- [Mike Love's general reference card](https://github.com/mikelove/bioc-refcard)
- [Motivations and core values (optional)](pages/biomotiv.html)
- [Installing Bioconductor and finding help](pages/installing_Bioconductor_finding_help.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/bioc/installing_Bioconductor_finding_help.Rmd)
- [Data structure and management for genome scale experiments](pages/dataman2019.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocintro_5x/dataman2019.Rmd)
    - [Coordinating multiple tables: ExpressionSet](pages/dataman2019.html#coord)
    - [Institutional archives: GEO, ArrayExpress](pages/dataman2019.html#geoquery)
- [Interlude: Working with general genomic features using GenomicRanges](pages/bioc1_igranges.html) 
    - [IRanges introduced](pages/bioc1_igranges.html#IRanges) 
    - [Intra-range operations](pages/bioc1_igranges.html#intrarange) 
    - [Inter-range operations](pages/bioc1_igranges.html#interrange) 
    - [GRanges](pages/bioc1_igranges.html#granges) 
    - [Calculating overlaps](pages/bioc1_igranges.html#findoverlaps) 
- [Range-oriented solutions for current experimental paradigms](pages/dataman2019.html#summexp)
    - [SummarizedExperiment: for RNA-seq and 450k methylation](pages/dataman2019.html#summexp)
    - [External storage for very large assays](pages/dataman2019.html#extern)
    - [GenomicFiles for families of BAM or BED](pages/dataman2019.html#genomicFiles)
    - [DNA Variants: VCF handling with VariantAnnotation and VariantTools](pages/dataman2019.html#variants)
    - [Handling multiomic archives like TCGA](pages/dataman2019.html#multiomic)
    - [Cloud-oriented solutions: e.g., Google BigQuery](pages/dataman2019.html#cloud)
- [Short read mapping/alignment software (optional)](pages/bioc1_align.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_align.Rmd)


## Chapter 12 - Genomic Annotation with Bioconductor

- [More details on GRanges](pages/bioc1_igranges.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_igranges.Rmd)
    - [Run-length encoding, views](pages/bioc1_igranges.html#rleviews) 
    - [Application to genomic landmarks](pages/bioc1_igranges.html#genomic) 
    - [Application to 450k methylation array visualization](pages/bioc1_igranges.html#methyviz) 
- [General overview of Bioconductor annotation](pages/bioc1_annoOverview.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_annoOverview.Rmd)
    - [Levels: reference sequence, regions of interest, pathways](pages/bioc1_annoOverview.html#threelev)
    - [Discovering reference sequence](pages/bioc1_annoOverview.html#findingref)
    - [A build of the human genome](pages/bioc1_annoOverview.html#hsap)
    - [Gene/Transcript/Exon catalogs from UCSC and Ensembl](pages/bioc1_annoOverview.html#txUCSCnENSEMBLE)
    - [Importing and exporting regions and scores](pages/bioc1_annoOverview.html#importExport)
    - [AnnotationHub: brokering thousands of annotation resources](pages/bioc1_annoOverview.html#annhub)
    - [OrgDb: simple interface to annotation databases](pages/bioc1_annoOverview.html#orgdb)
    - [Finding and managing gene sets](pages/bioc1_annoOverview.html#genesets)
    - [OrganismDb: unifying diverse annotation](pages/bioc1_annoOverview.html#organismdb)
- [Cheat sheet on Bioconductor annotation](pages/bioc1_annoCheat.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_annoCheat.Rmd)
- [Translating addresses between genome builds: liftOver](pages/bioc1_liftOver.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_liftOver.Rmd)

## Chapter 13 - Genome-scale hypothesis testing with Bioconductor

- [Distinguishing biological and technical variability](pages/bioc1_btvari.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_btvari.Rmd)
    - [An experiment with pooled and individual samples](pages/bioc1_btvari.html#pooling) 
    - [Measuring technical variation](pages/bioc1_btvari.html#techContrast) 
    - [Measuring biological variation](pages/bioc1_btvari.html#biovar) 
    - [Interpretation](pages/bioc1_btvari.html#interpretation) 
- [Multiple comparisons with genewise t-tests](pages/bioc1_t_mult.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_t_mult.Rmd)
    - [Gene-wise testing](pages/bioc1_t_mult.html#rowWiseT)
    - [Naive enumeration of genes](pages/bioc1_t_mult.html#naive)
    - [Demonstrating danger of multiple testing with a set of sham comparisons](pages/bioc1_t_mult.html#sham)
    - [Adjusting for multiplicity with qvalue](pages/bioc1_t_mult.html#adjusted)
    - [Adjusted counts in the sham case](pages/bioc1_t_mult.html#demo)
- [Moderated t tests via limma](pages/bioc1_limma.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_limma.Rmd)
    - [A spike-in dataset](pages/bioc1_limma.html#spikein)
    - [Naive t-tests](pages/bioc1_limma.html#simpleT)
    - [Three steps with limma: lmFit, eBayes, topTable](pages/bioc1_limma.html#limmaSteps)
    - [Exposing the spiked-in genes](pages/bioc1_limma.html#expose)
    - [A view of the shrinkage of variance estimates](pages/bioc1_limma.html#shrinkage)
- [Introducing gene sets and gene set analysis](pages/bioc1_geneset_1.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocintro_5x/bioc1_geneset_1.Rmd)
    - Data wrangling
        - [A dataset for comparing expression by gender](pages/bioc1_geneset_1.html#dataprep)
        - [Finding surrogate variables/batch effect correction](pages/bioc1_geneset_1.html#batch)
    - [The Broad Institute MsigDb](pages/bioc1_geneset_1.html#MSIGDB)
        - [Identifier remapping](pages/bioc1_geneset_1.html#identifierMapping)
        - [Categorical testing](pages/bioc1_geneset_1.html#chisq)
        - [Statistical summaries for sets: Wilcoxon](pages/bioc1_geneset_1.html#wilcox)
        - [Statistical summaries for sets: t statistics](pages/bioc1_geneset_1.html#ttest)
    - [Adjusting for within-set correlation](pages/bioc1_geneset_1.html#correl)
    - [A permutation procedure](pages/bioc1_geneset_1.html#permutations)


## Chapter 14 - Visualization of genome scale data

- [A basic overview of visualization tasks and strategies](pages/bioc2_vizOv.html)[[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_vizOv.Rmd)
    - [Gene models](pages/bioc2_vizOv.html#genemods)
    - [Gene models plus data](pages/bioc2_vizOv.html#genemodsData)
    - [Driving visualizations with functions](pages/bioc2_vizOv.html#vizFunc)
    - [Using the browser to drive visualization functions via shiny](pages/bioc2_vizOv.html#vizShiny)
    - [Queriable dynamic displays with plotly](pages/bioc2_vizOv.html#vizPlotly)
- Annotation-oriented visualizations
    - [Sketching the binding landscape over chromosomes with ggbio's karyogram layout](pages/bioc2_ggbio.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_ggbio.Rmd)
    - [Plotting data in the context of genomic features with Gviz](pages/bioc2_gvfeat.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_gvfeat.Rmd)
- [Visualizing NGS data](pages/bioc2_vizNGS.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_vizNGS.Rmd)
- Interactive visualization
    - [Graphical user interfaces for multivariate data with shiny](pages/bioc2_shiny.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_shiny.Rmd)
    - [Clustering gene expression data with shiny](pages/esHclust.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/esHclust.Rmd)
- [Final remarks on visualization](pages/finalViz.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/finalViz.Rmd)

## Chapter 15: Pursuing scalability in genomic analysis: parallelism and out-of-memory data

- [Parallel computing with R and Bioconductor](pages/bioc2_parallel.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_parallel.Rmd)
    - [Demonstrating simple speedup in multicore environments](pages/bioc2_parallel.html#speedup)
    - [Implicit parallelism with BiocParallel and GenomicAlignments](pages/bioc2_parallel.html#implicit)
- [External data: data interfaces that spare RAM](pages/bioc2_externData.html)[[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_externData.Rmd)
    - [SQLite for annotation](pages/bioc2_externData.html#sqlite)
    - [Tabix-indexed BAM](pages/bioc2_externData.html#tabix)
    - [HDF5](pages/bioc2_externData.html#hdf5)
    - [An illustration of NoSQL with S4: mongodb and RaggedMongoExpt](pages/bioc2_nosql.html)[[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_nosql.Rmd)
- [Benchmarking various out-of-memory solutions](pages/multiOOM.html)[[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/multiOOM.Rmd)
- [Introduction to Bioconductor's Amazon Machine Instance for cluster creation and use in EC2](pages/bioc2_HPCami.html) [[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_HPCami.Rmd)
- [Sharded GRanges for scalable integrative analysis](pages/bioc2_hybstor.html)[[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_hybstor.Rmd)


### Chapter 16: Multi-omic data integration 

- [Basic examples of multi-omic integration](pages/bioc2_integExamps.html)[[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_integExamps.Rmd)
    - [Transcription factor (TF) binding and gene coexpression in yeast](pages/bioc2_integExamps.html#yeasttfs)
    - [TF binding and GWAS hits in humans](pages/bioc2_integExamps.html#gwastf)
- [Using RTCGAToolbox outputs to integrate clinical, mutation, expression and methylation assays](pages/tcga.html)[[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/tcga.Rmd)
    - [Basic data acquisition](pages/tcga.html#firehose)
    - [Working with clinical data](pages/tcga.html#clinical)
        - [Defining a severity marker](pages/tcga.html#severity)
        - [Extracting survival times](pages/tcga.html#survival)
    - [Working with mutations](pages/tcga.html#mutation)
    - [Curation tasks for discrepant identifier formats](pages/tcga.html#curation)
    - [Working with expression data](pages/tcga.html#expression)
        - [Associating tumor stage with expression patterns](pages/tcga.html#tstageToExpr)
        - [Linking DNA methylation with expression patterns](pages/tcga.html#methy)
- [Application to visualization: kataegis and rainfall plot](pages/bioc2_rainfall.html)[[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_rainfall.Rmd)

### Chapter 17: Fostering reproducible genome-scale analysis 

- [Overview of unit on reproducibility](pages/bioc2_repro1.html)[[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_repro1.Rmd)
    - [Basic definitions](pages/bioc2_repro1.html#defs)
    - [Infrastructure requirements](pages/bioc2_repro1.html#infrastr)
    - [Statistical aspects of reproducibility](pages/bioc2_repro1.html#infrastr)
    - [Analysis of reproducibility probability (Boos and Stefanski 2011)](pages/bioc2_repro1.html#RP)
    - [Costs of highly reproducible designs](pages/bioc2_repro1.html#costs)
- [Package structure, creation, installation, management](pages/bioc2_rpacks.html)[[Rmd]](https://github.com/genomicsclass/labs/blob/master/biocadv_6x/bioc2_rpacks.Rmd)
    - [What is a package?](pages/bioc2_rpacks.html#defpack)
    - [Using `package.skeleton`](pages/bioc2_rpacks.html#skel)
    - [Using `makeOrganismPackage`](pages/bioc2_rpacks.html#org)
    - [Using `devtools`](pages/bioc2_rpacks.html#devtools)
        - [`create()` to set up folders and DESCRIPTION](pages/bioc2_rpacks.html#create)
        - [Composing documentation plus code](pages/bioc2_rpacks.html#source)
        - [`document()`, `install()`](pages/bioc2_rpacks.html#docinst)
    - [Conclusions, including a link to a recent Nature Toolbox article on Bioconductor](pages/bioc2_rpacks.html#wrapup)


---

## Legacy material from 2015 Introduction to Bioconductor

- [Installing Bioconductor and finding help](pages/installing_Bioconductor_finding_help.html)
- [Annotating phenotypes and molecular function](pages/annoPhen.html)
- [The ExpressionSet Container](pages/eset.html)


- [IRanges and GRanges](pages/iranges_granges.html)
- [Operating on GRanges](pages/operateGRanges.html)


- [Cheat sheet for genomic annotation](pages/annoCheat.html)
- [Translating addresses between reference builds with liftOver](pages/anno4liftover.html)
- [Cheat sheet for GRanges and other Bioconductor objects](https://github.com/mikelove/bioc-refcard/blob/master/README.Rmd)


- [Importing NGS data with Bioconductor](pages/import_NGS.html)
- [NGS read counting](pages/read_counting.html)


- [Technical versus biological variability](pages/biological_versus_technical_var.html)
- [Statistical Inference with Bioconductor](pages/inference_with_bioc.html)
- [Using limma](pages/using_limma.html)
- [Gene Sets Analysis](pages/gene_set_analysis.html)
- [Gene Sets Analysis in R](pages/gene_set_analysis_in_R.html)


<!--

new ones:
- [Computing with chromosomes and variants](pages/chromComp.html)

old ones:
- [Reading microarray data](pages/reading_microarray_data.html)
- [Downloading data from GEO using GEOquery](pages/GEOquery.html)
- [Background](pages/background.html)
- [Normalization](pages/normalization.html)
- [SVA, ComBat, PCA and SVD](pages/svacombat.html)
- [Basic inference for microarray](pages/basic_inference_microarray.html)
- [EDA plots for microarray](pages/EDA_plots_for_microarray.html)
- [EDA plots for next generation sequencing](pages/EDA_plots_for_NGS.html)
- [Visualizing NGS data](pages/visualizing_NGS.html)
- [Mapping features to genes](pages/mapping_features.html)
- [Gene set analysis lecture](pages/gsa.html)
- [Gene set testing in R](pages/gene_set_testing.html)

-->

## RNA-seq data analysis

- [Downloading and unzipping fastq files](https://github.com/genomicsclass/labs/blob/master/rnaseq/fastq.md)
- [Genomic alignment with STAR](https://github.com/genomicsclass/labs/blob/master/rnaseq/genome_align_STAR.md)
- [Transcriptome alignment with RSEM](https://github.com/genomicsclass/labs/blob/master/rnaseq/trancsript_align_RSEM.md)


- [RNA-seq at the gene-level: EDA, DE and SVA](pages/rnaseq_gene_level.html)
- [Differential exon usage](pages/rnaseq_exon_usage.html)
- [Exploring plots of isoform-level expression with Cufflinks/cummeRbund](pages/rnaseq_isoform_cummerbund.html)

## Variant Discovery and Genotyping

- [Genome variation](pages/SNP.html) from 2014

## ChIP-seq data analysis

- [ChIP-seq](pages/ChIPseq.html) from 2014

## DNA methylation data analysis

- [DNA Methylation Data Analysis](pages/methylation.html)
- [Reading 450K idat files with the minfi package](pages/minfi.html)
- [Interactive visualization of DNA methylation data analysis](pages/epiviz.html)
 
 
 - [Statistical Inference in the Analysis of DNA methylation Data](pages/inference_for_DNAmeth.html)

---

## Footnotes for all lectures

- [Footnotes](pages/footnotes.html)

## Acknowledgments

- [Course acknowledgments](pages/acknowledgments.html)

